src/layouts/AccessWalletLayout/components/HardwarePasswordModal/HardwarePasswordModal.vue

```diff
@@ -78,12 +80,32 @@ export default {
       this.password == '';
       this.$refs.passwordInput.focus();
     },
-    unlockWallet() {
-      this.walletConstructor('', this.password)
-        .then(_newWallet => {
-          this.$emit('hardwareWalletOpen', _newWallet);
-        })
-        .catch(this.walletConstructor.errorHandler);
+    async unlockWallet() {
+      if (this.hardwareBrand === 'CoolWallet') {
+        await cwsTransportLib.listen(async (error, device) => {
+          if (device) {
+            const transport = await cwsTransportLib.connect(device);
```

Comment:
this should be moved inside the wallet init function? I think previous `unlockWallet` function should be sufficient

Reply:
moving it here fixed the scoping issue